client/src/App.jsx

```diff
@@ -84,6 +90,30 @@ export default function App() {
         handleGameRestart()
     }
 
+   
+
+    function checkImageDuplicates(data){
+        const picList = {}
+        const result = []
+
+        data.every((x,i) => {
+            if(result.length==numOfRandomUsers){
+                return false
+            }
+            if (x.picture.large in picList){
+                console.log("dup found")
+            }else{
+            result.push(x)
+            picList[x.picture.large] = 1
+            }
+
+            return true
+
+        })
+        
+        return result
+    }
```

Comment:
this function is a little confusing. i understand what it does but it takes time to read it and understand. i think you can clean up the code here and make it more readable. 

- first of all "checkimageDuplicate" is a lie because the function doesnt just check it, it seems to be returning an array after doing some filtering so its clearly not just "checking". so i would suggest a better function name like "removeDuplicateUsers" ?. 

- how come you used the "every" method. i know you are using that as your loop control but it seems like a uncommon way to do it. maybe you can explain but to me its confusing. can you not just use a simple for loop and then "break" if you reach the desired length? 
- also why do we check for "picture.large"? can we not check the image url? cus what if in the future we change from large to something else. this code will need to be updated too which seems like something easy to miss. 
- how come picList is a dict? i think ... i get what youre doing to keep track of the distinct images... but its confusing. i think you can use a different data structure here like a set?
- you can probably use better variable names instead of "result" and "picList". result is just your uniqueUsers. and no reason to include "list" in your variable name because thats the datatype. but you can probably call that something else too like "pics" or "UserImages"

Reply:
i was thinking about this and i see that using the every method is a more declarative way of doing this. and react is mainly declarative. but i would still argue that in this case because it takes away from readability you will be better off using a traditional for loop style which is okay.

Reply:
So theres no real reason im using every i hear prefer to see x instead of x[i] cuz it's cleaner. I can switch it to a forloop if u think it'll look cleaner

Reply:
I'm using the dict just cuz ik its quick and i was already familiar with the syntax. You are right that using a set is a better data structure for this case so ill swap it out for that as well

Reply:
Picture.large contains the image url which is what I'm checking to see if that repeats

Reply:
Ack on the naming conventions ill change it up and add some documentation on the function for clarity